core/src/main/scala/chisel3/Aggregate.scala

```diff
@@ -963,6 +963,10 @@ abstract class Bundle(implicit compileOptions: CompileOptions) extends Record {
     for (m <- getPublicFields(classOf[Bundle])) {
       getBundleField(m) match {
         case Some(d: Data) =>
+          if (d.binding.nonEmpty) {
+            throwException(s"Bundle can only take type elements (got value binding '$d' on '${m.getName()}')")
+          }
```

Comment:
Just another idea, maybe `requireIsChiselType` can catch this?

Reply:
I think this suggestion conveys the intent a little more clearly.